Kudu.Services/Deployment/DeploymentController.cs

```diff
@@ -79,17 +85,33 @@ public void Delete(string id)
         /// </summary>
         /// <param name="id">id of the deployment to redeploy</param>
         [HttpPut]
-        public async Task Deploy(string id = null)
+        public async Task<HttpResponseMessage> Deploy(string id = null)
         {
             JObject result = GetJsonContent();
 
             // Just block here to read the json payload from the body
             using (_tracer.Step("DeploymentService.Deploy(id)"))
             {
+                HttpResponseMessage response = Request.CreateResponse(HttpStatusCode.OK);
                 await _deploymentLock.LockHttpOperationAsync(async () =>
                 {
                     try
                     {
+                        DeployResult deployResult;
+                        if (TryParseDeployResult(id, result, out deployResult))
+                        {
+                            using (_tracer.Step("DeploymentService.Create(id)"))
+                            {
+                                CreateDeployment(deployResult, result.Value<string>("details"));
+
+                                deployResult.Url = Request.RequestUri;
+                                deployResult.LogUrl = UriHelper.MakeRelative(Request.RequestUri, "log");
+
+                                response = Request.CreateResponse(HttpStatusCode.OK, ArmUtils.AddEnvelopeOnArmRequest(deployResult, Request));
+                                return;
+                            }
+                        }
+
                         if (_autoSwapHandler.IsAutoSwapOngoing())
```

Comment:
right, it is out of scope of this PR. since the logic has been working and we do not have issue with it. i think we are good. just curios about it.

:shipit: